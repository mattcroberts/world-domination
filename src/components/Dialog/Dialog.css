.Dialog {
  display: none;
  background: lightblue;
  position: absolute;
  z-index: 1;
  width: 50%;
  height: 30%;
}

.Dialog.show {
  display: block;
}
