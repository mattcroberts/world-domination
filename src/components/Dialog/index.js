import React from 'react';
import classnames from 'classnames';

import './Dialog.css';

export default ({ visible }) => (
    <form className={classnames('Dialog', { show: visible })} />
);
