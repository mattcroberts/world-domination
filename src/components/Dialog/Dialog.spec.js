import { shallow } from 'enzyme';
import React from 'react';

import Dialog from './index';

describe('Dialog', () => {
    it('should render', () => {
        const component = shallow(<Dialog />);

        expect(component).toMatchSnapshot();
    });

    it('should show when visible prop is true', () => {
        const component = shallow(<Dialog visible={true} />);
        expect(component).toMatchSnapshot();
        expect(component.find('form').hasClass('show')).toEqual(true);
    });

    it('should hide when visible prop is false', () => {
        const component = shallow(<Dialog visible={false} />);
        expect(component.find('form').hasClass('show')).toEqual(false);
    });
});
