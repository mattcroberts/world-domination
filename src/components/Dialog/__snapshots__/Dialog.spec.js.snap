// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render 1`] = `
<form
  className="Dialog"
/>
`;

exports[`Dialog should show when visible prop is true 1`] = `
<form
  className="Dialog show"
/>
`;
